### PR TITLE
Change malloc to calloc

### PR DESCRIPTION
**Pull request by agorshk**

### CODE REVIEW

**Comment by agorshk** [comment]
make test

**Comment by agorshk** [comment]
make test

**Comment by agorshk** [comment]
make test

**Comment by agorshk** [comment]
CI was rerun, id=e9e52b3f-c43b-f1f2-a863-8cdcd4b723a1

File: include/services/collection.h [hunk 1]
```diff
@@ -228,7 +228,7 @@ class Collection
     bool resize(size_t newCapacity)
     {
         if(newCapacity <= _capacity) { return true; }
-        T *newArray = (T *)services::daal_malloc(sizeof(T) * newCapacity);
+        T *newArray = (T *)services::daal_calloc(sizeof(T) * newCapacity);
```
Comment (ShvetsKS) [inline include/services/collection.h]: does it affect performance?

Comment (agorshk) [reply include/services/collection.h]: This changes were checked in some algorithms and performance is ok, but full performance results might be different.

**Pull request merged**